packages/plugin-pnp/sources/PnpLinker.ts

```diff
@@ -18,6 +18,17 @@ const FORCED_UNPLUG_PACKAGES = new Set([
   structUtils.makeIdent(null, `fsevents`).identHash,
 ]);
 
+const FORCED_UNPLUG_FILETYPES = new Set([
+  // Windows can't execute exe files inside zip archives
+  '.exe',
+  // The c/c++ compiler can't read files from zip archives
+  '.h', '.hh', '.hpp', '.c', '.cc', '.cpp',
```

Comment:
```suggestion
  '.h', '.hh', '.hpp', '.c', '.cc', '.cpp',
  // iOS (react-native)
  '.m', '.mm', '.swift',
  // Android (react-native)
  '.kt',
```

React-native is a popular node/yarn environment, and react-native often comes with native code. These should be unplugged as well.

Reply:
I just realized this was a PR from a year ago, so I don't expect anybody to see the above comment!

Reply:
You'd be surprised 🙂 I'm not expert in RN but that seems reasonable enough, feel free to open a PR 👍

Reply:
@arcanis thanks for the feedback!

 I checked up on the main (`master`) branch and found this: https://github.com/yarnpkg/berry/blob/6b840bcdd2df0e91e09b515a1d075e0e3b934a70/packages/plugin-pnp/sources/PnpLinker.ts#L363-L375

Which appears to not use a file extension system at all anymore. I think packages have to add a `preferUnplugged` option to my package.json files, but that's not super convenient for 3rd party packages I don't control :(

https://next.yarnpkg.com/configuration/manifest#preferUnplugged

Reply:
Seems reasonable indeed, PR welcome :)

Reply:
@fbartho it got moved here: https://github.com/yarnpkg/berry/blob/4f72425f91ea3f110f99f17aa2132795b56beb78/packages/plugin-pnp/sources/jsInstallUtils.ts#L65-L74